main.py

```diff
@@ -106,10 +106,47 @@ def get_name_tag(resource_type, name):
     ]
 )
 
+# Create EC2 for public subnets
+instances = []
+for i, public_subnet in enumerate(public_subnets):
+    instance = ec2.create_instances(
+        ImageId='ami-0c0b74d29acd0cd97',
+        InstanceType='t2.micro',
+        MaxCount=1,
+        MinCount=1,
+        SubnetId=public_subnet.id,
+        SecurityGroupIds=[public_sg.id],
+        TagSpecifications=get_name_tag('instance', f'boto3-public-instance-{i + 1}')
+    )[0]  # ec2.create_instances() returns a list of instances, we take the first one
+    instances.append(instance)
+
+    instance.wait_until_running()
+
+# Create EC2 for private subnet
+for i in range(2):
+    ec2.create_instances(
+        ImageId='ami-0c0b74d29acd0cd97',
+        InstanceType='t2.micro',
+        MaxCount=1,
+        MinCount=1,
+        SubnetId=private_subnet.id,
+        SecurityGroupIds=[private_sg.id],
+        TagSpecifications=get_name_tag('instance', f'boto3-private-instance-{i + 1}')
+    )
+
 
 # Deleting the built infrastructure
 input('Press Enter to destroy the infrastructure')
 
+instances = ec2.instances.filter(
```

Comment:
Since we already have the list of instances, we won't need to filter all instances in the region. It's worth noting that this isn't filtering just the instances in the list but all instances in the region, meaning that all running and stopped instances will be terminated, not just those created by boto3.